atomic

```diff
@@ -683,6 +682,12 @@ def create_parser(help_text):
     verifyp.add_argument("-v", "--verbose", default=False,
                           action="store_true",
                           help=_("Report status of each layer"))
+    verifyp.add_argument("-V", "--validate", default=True,
+                                action="store_false",
+                                help=_("disable validating system images"))
+    verifyp.add_argument("-g", "--generate", default=False,
```

Comment:
Also, the immutable bit can be removed.  With O_OBJECT at least you'd have to have compromised the kernel (or have CAP_SYS_ADMIN to write to the raw disk).

In the bigger picture of course if someone has `CAP_SYS_ADMIN` on your machine, they can inject code into the kernel to lie to whatever userspace validation tools you have, or things like [inject malware into the hard drive controller](http://hackaday.com/2015/06/08/hard-drive-rootkit-is-frighteningly-persistent/).

That said, if we did have the mtree data in OCI images we built, one could use them to more efficiently perform offline forensics too.  Mount a disk with libguestfs or whatever, download just the mtree metadata from a layer rather than the whole layer tar.